pkg/gameserversets/controller.go

```diff
@@ -618,6 +619,15 @@ func computeStatus(list []*agonesv1.GameServer) agonesv1.GameServerSetStatus {
 		case agonesv1.GameServerStateReserved:
 			status.ReservedReplicas++
 		}
+
+		// Aggregates all Counters and Lists only for GameServer states Ready, Reserved, or Allocated.
+		if runtime.FeatureEnabled(runtime.FeatureCountsAndLists) &&
+			(gs.Status.State == agonesv1.GameServerStateReady ||
```

Comment:
(non blocking): Could also make this a function much like this one:

https://github.com/googleforgames/agones/blob/adfb30845d1b06cf3a04d84f2bbd19f128056254/pkg/apis/agones/v1/gameserver.go#L686

i feel like we might have this kind of check in a few places 🤔 ?

Reply:
I added in an "IsReady" function to gameserver.go. Let me know if you can think of a better name for this func.